### PR TITLE
Fix bad seeking behavior in MMCMP compressed blocks.

### PR DESCRIPTION
**Pull request by AliceLR**
Fixes one of the issues in #517: files ending with an 8-bit compressed block shorter than 256 bytes would be rejected because the ptable handling was incorrectly adapted from libmodplug.

### CODE REVIEW

**Comment by AliceLR** [comment]
Ugh, MikMod actually does try to load the entire 0x100 bytes as well. Surely nothing *actually* relies on table entries past the bound, right? Thanks, libmodplug...

**Comment by AliceLR** [comment]
Revised to be closer to Olivier Lapicque's, uhhhh, vision. 🤦‍♀️

File: src/depackers/mmcmp.c [hunk 1]
```diff
@@ -232,8 +232,13 @@ static int block_unpack_8bit(struct block *block, struct sub_block *sub,
 	uint32 numbits = block->num_bits;
 	uint32 j, oldval = 0;
 	uint8 ptable[0x100];
+	long seekpos = hio_tell(in) + block->tt_entries;
 
-	if (hio_read(ptable, 1, 0x100, in) != 0x100) {
+	/* The way the original libmodplug depacker is written allows values
+	 * to be read from the compressed data. It's impossible to tell if this
+	 * was intentional or yet another bug. Nothing seems to rely on it. */
+	memset(ptable, 0, sizeof(ptable));
```
Comment (sezero) [inline src/depackers/mmcmp.c:240]: do we need the memset ?

Comment (AliceLR) [reply src/depackers/mmcmp.c:240]: Yes. The input stream can index any position in that array, though valid data *seems* to never index beyond `block->tt_entries`. The read is not guaranteed to initialize the entire array (which is why EVERSMOK.S3M failed to load).

Comment (sezero) [reply src/depackers/mmcmp.c:240]: >  (which is why EVERSMOK.S3M failed to load)

The thing is, I can load and play EVERSMOK.S3M with that memset commented out

Comment (AliceLR) [reply src/depackers/mmcmp.c:240]: That's because before the patch it would reject the module unless the read returned 0x100.

Comment (sezero) [reply src/depackers/mmcmp.c:240]: OK then.

**Review by sezero — APPROVED** [review]
This is good for one of the failures, so let's apply.

I guess we go back to malloc instead calloc for `outbuf.buf` and
continue investigating.

**Pull request merged**